cg/models/downsample/downsample_meta_data.py

```diff
@@ -0,0 +1,167 @@
+"""Model for down sampling meta data."""
+import logging
+from pathlib import Path
+
+from cg.constants import Priority, SequencingFileTag
+from cg.models.cg_config import CGConfig
+from cg.store.models import ApplicationVersion, Family, Sample
+
+LOG = logging.getLogger(__name__)
+
+
+class DownsampleMetaData:
+    def __init__(
+        self, config: CGConfig, sample_internal_id: str, number_of_reads: int, case_internal_id: str
+    ):
+        """Initialize the model."""
+        self.config = config
+        self.sample_internal_id: str = sample_internal_id
+        self.number_of_reads: int = number_of_reads
+        self.case_internal_id: str = case_internal_id
+        self.original_sample: Sample = self.get_sample_to_downsample()
+        self.original_case: Family = self.get_case_to_downsample()
+        self.has_enough_reads: bool = self.has_enough_reads_to_downsample()
+        self.downsampled_sample: Sample = self._generate_statusdb_downsampled_sample_record()
+        self.downsampled_case: Family = self._generate_statusdb_downsampled_case()
+        self.create_down_sampling_working_directory()
+        LOG.info(f"Pre-flight checks completed for {self.sample_internal_id}")
```

Comment:
I don't think all this should be done in the instantiation of the model. What is the reasoning for putting it there?

Reply:
To me it makes the code more compact. Since the class organises itself upon initialising it. It would avoid needed to add extra code to call and update. Why do you think it should not be done like this?